feature(treewide): Use simulated racks globally

pehala opened the pull request:
4 node tests were switched to 6 nodes test and sometimes I decreased the instance type

Fixes https://github.com/scylladb/scylla-cluster-tests/issues/10241

### Testing
<!-- Add links to Argus/Jenkins of test test done with this PR -->
<!-- This would help the reviewer to cross check what was tested, and and review the results as needed -->
- [x] None, please let me know what tests you would like me to run.
   - Given that we are already running Tier 1 and Sanities with these changes, we have at least some confidence

### PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [x] I added the relevant `backport` labels
   - Backports will be done for 2025.1 and perf branches, but I will do them only after we tested it sufficiently in master  
- [x] I didn't leave commented-out/debugging code

pehala commented:
I had to update unit tests, even though no logic was changed. This is yet another example of our unittests not being separated enough and heavily relying on actual production configuration. It is a similar problem to https://github.com/scylladb/scylla-cluster-tests/issues/10592, which I believe should be tacked as well

pehala commented:
@mikliapko Given this PR will change how racks/dcs are named, did you already upload new snapshots with new names?

mikliapko commented:
> @mikliapko Given this PR will change how racks/dcs are named, did you already upload new snapshots with new names?

Please, give me one-two days more. 
The implementation is not here yet, was busy with 1-1 restore tasks.

pehala commented:
> Please, give me one-two days more.
> The implementation is not here yet, was busy with 1-1 restore tasks.

No problem, let me know if I can help in any way

pehala commented:
@scylladb/qa-maintainers Do you have suggestions for job that need testing?

pehala commented:
> ### identified changes in generated code
> 
> diff found by running: `bash ./docker/env/hydra.sh update-conf-docs`:
> 
> ```
>  docs/configuration_options.md | 2 +-
>  1 file changed, 1 insertion(+), 1 deletion(-)
> ```

Errors was not introduced here, created separate fix https://github.com/scylladb/scylla-cluster-tests/pull/10960

mikliapko reviewed (commented):
I suppose putting `simulated_racks: 0` into `test-cases/manager/manager-regression-multiDC-set-distro.yaml` should not change the current flow. If you can confirm it without testing, that's nice, otherwise, I'd like to ask you to run [ubuntu22-sanity-test](https://jenkins.scylladb.com/job/manager-master/job/ubuntu22-sanity-test/) job just to be sure.

pehala commented:
> I suppose putting `simulated_racks: 0` into `test-cases/manager/manager-regression-multiDC-set-distro.yaml` should not change the current flow. If you can confirm it without testing, that's nice, otherwise, I'd like to ask you to run [ubuntu22-sanity-test](https://jenkins.scylladb.com/job/manager-master/job/ubuntu22-sanity-test/) job just to make sure.

Done, it passed

pehala commented:
@scylladb/qa-maintainers I think this PR is ready, I run all requested tests (+ some privately) so this should be ready for merge. It can still cause issues but when it happens we will fix the cases one by one

pehala commented:
> Simulated racks doesn't affect operator deployments.

The code should not affect it, but I added it to the defaults just to be sure as well

> Node count and instance type changes do affect in unexpected way.

Reverted

vponomaryov commented on defaults/k8s_local_kind_config.yaml:
```diff
@@ -42,3 +42,5 @@ backtrace_decoding: false
 docker_image: ''
 backup_bucket_location: 'minio-bucket'
+
+simulated_racks: 0
```
This is redundant change.

vponomaryov commented on defaults/k8s_gke_config.yaml:
```diff
@@ -67,3 +67,5 @@ append_scylla_args: '--abort-on-lsa-bad-alloc 1 --abort-on-internal-error 1 --ab
 backup_bucket_backend: 's3'
 backup_bucket_location: 'minio-bucket'
+
+simulated_racks: 0
```
This is redundant change.

vponomaryov commented on defaults/k8s_eks_config.yaml:
```diff
@@ -55,3 +55,5 @@ k8s_instance_type_auxiliary: 't3.large'
 append_scylla_args: '--abort-on-lsa-bad-alloc 1 --abort-on-internal-error 1 --abort-on-ebadf 1 --enable-sstable-key-validation 1'
 docker_image: ''
+
+simulated_racks: 0
```
This is redundant change.

pehala commented:
v4:
  * Remove `simulated_racks: 0` from k8s

vponomaryov commented on test-cases/performance/perf-regression-2mv.yaml:
```diff
@@ -10,11 +10,11 @@ stress_cmd_no_mv_profile: 'data_dir/cs_no_mv_basic_profile.yaml'
 #    profile: 'data_dir/cs_mv_basic_profile_by_firstname_id.yaml'
 
-n_db_nodes: 4
+n_db_nodes: 6
```
This is configuration for a performance test.
Changing resource set in a performance tests is crucial.

But, I am not sure anyone uses this job config.

@fruch , @soyacz is it used anywhere?

vponomaryov reviewed (approved):
LGTM

fruch commented on test-cases/longevity/longevity-alternator-3h-multidc.yaml:
```diff
@@ -31,9 +31,12 @@ round_robin: true
-region_aware_loader: true
 n_monitor_nodes: 1
 
+rack_aware_loader: true
```
This should be removed, it's irrelevant for this test...

fruch commented:
@pehala 

first fallout of this one:
https://argus.scylladb.com/tests/scylla-cluster-tests/3c1828ff-3174-43a4-aea7-27d597e06f18

yaronkaikov commented:
@fruch @pehala seems this patch failed docker artifacts - https://jenkins.scylladb.com/job/scylla-master/job/artifacts/job/artifacts-docker-test/1398/

please revert or submit a fix ASAP
